service/controller/v3/resource_set.go

```diff
@@ -299,6 +300,11 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 	}
 
 	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
+		subnets, err := network.ComputeFromCR(ctx, obj)
```

Comment:
I'd preffer asserting `obj` type here before calling `network.ComputeFromCR` and making `network.ComputeFromCR` argument strongly typed.

Reply:
done